staging/src/k8s.io/api/batch/v1/types.go

```diff
@@ -535,6 +535,23 @@ const (
 	JobFailureTarget JobConditionType = "FailureTarget"
 )
 
+type JobReasonType string
+
+const (
+	// PodFailurePolicy reason indicates a job failure condition is added due to
+	// a failed pod matching a pod failure policy rule
+	// https://kep.k8s.io/3329
+	// This is currently a beta field.
+	PodFailurePolicyMatched JobReasonType = "PodFailurePolicy"
+	// BackOffLimitExceeded reason indicates that pods within a job have failed a number of
+	// times higher than backOffLimit times.
+	BackoffLimitExceeded JobReasonType = "BackoffLimitExceeded"
+	// DeadlineExceeded means job duration is past ActiveDeadline
+	DeadlineExceeded JobReasonType = "DeadlineExceeded"
+	// FailedIndexes means Job has failed indexes.
+	FailedIndexes JobReasonType = "FailedIndexes"
```

Comment:
suggest prefixing the constant names with `JobReason_` and documenting these are used as reasons in conditions on jobs (and the specific condition `type` these are used in, if possible).

as an example to follow:
https://github.com/kubernetes/kubernetes/blob/b8ef1a0791711f96e8dea05ad9ea026d351d67f3/staging/src/k8s.io/api/core/v1/types.go#L2871-L2892

Reply:
I think I did this as we do comment on the reasons in staging.